### PR TITLE
Backend add notification

### LINKED ISSUE #382: Implement Endpoint to GET Notifications For User
| path                               | type   | functionality                               | parameters                                                                                                                                                                                                                | example                                                                                                                                                                                                                                  |
|---------------------------------------------------------------|--------|---------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| notifications | GET | get user notifications |    |  |

The notification types can be found on [1.2.2 Notifications Requirements](https://github.com/bounswe/2021SpringGroup6/wiki/Project-Requirements#12-system-requirements).
These notifications should be added to the database on related functions.

For 1.2.2.2 It is decided that the few spots correspond to 10% of the total capacity.

### LINKED ISSUE #384: Create Notification Database
The notification database should have the following fields:
|    event_id   | user_id | type | read | date                                                                                                                                                                                                                                  |
|---------------------------------------------------------------|--------|---------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| | |  |    |  |

event_id : id of the related event
user_id: id of the user receiving notification
read: whether notification is seen by the user
date: date of the notification
type: one of the following notification types
 - user is approved for an event
 - user is rejected for an event
 - few spots left for interested event
 - all spots taken on participating event
 - event starts in a week, a day, and 3 hours.
 - event is cancelled
 - event is updated

### LINKED ISSUE #385: Implement Endpoint For Read Notifications
| path                               | type   | functionality                               | parameters                                                                                                                                                                                                                | example                                                                                                                                                                                                                                  |
|---------------------------------------------------------------|--------|---------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| notifications/<notification_id> | POST | Set notification as read |   | |

When the user reads a notification on front-end. This endpoint should be used to set the notification as read.

### PR DESCRIPTION
**Pull request by berkatil**
Closes #384 
Closes #382 
Closes #385

### CODE REVIEW

**Comment by berkatil** [comment]
I will add notifications for 1 week, 3 days, 1 day later

**Comment by berkatil** [comment]
I realized that I forgot to add notification for requirement 1.3.5.4(if the field capacity is less than number of spectators, notification should be sent about this). I will implement it, so let's not merge this branch until I implement it

**Comment by berkatil** [comment]
> I will add notifications for 1 week, 3 days, 1 day later

I have added this

**Comment by berkatil** [comment]
> I realized that I forgot to add notification for requirement 1.3.5.4(if the field capacity is less than number of spectators, notification should be sent about this). I will implement it, so let's not merge this branch until I implement it

I have implemented this as well, so it is ready to review

File: back-end/sports_platform/sports_platform_api/helpers/notification_creation.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            "description": f"The event with name {notification.event_id.name} is full now.",
+            "notification_id": notification.id,
+            "event_id": notification.event_id.event_id,
+            "date": notification.date,
+            "type": notification.notification_type
+    }
+
+def _event_accept(notification):
+    return {
+            "description": f"You are accpeted for the event with name {notification.event_id.name}",
```
Comment (fileames) [inline back-end/sports_platform/sports_platform_api/helpers/notification_creation.py:12]: There is a small typo here. "accpeted"

Comment (berkatil) [reply back-end/sports_platform/sports_platform_api/helpers/notification_creation.py:12]: thanks for the remark

**Comment by fileames** [comment]
Thanks for the effort.

**Pull request merged**